### PR TITLE
Servlet 2, Akka Http, Vert.x testing - Migrate to HttpServerTest

### PR DESCRIPTION
**Pull request by tylerbenson**
Adds `ignoreActiveSpan()` to each instrumentation.

### CODE REVIEW

File: dd-java-agent/instrumentation/akka-http-10.0/src/test/groovy/AkkaHttpServerInstrumentationTest.groovy [hunk 1]
```diff
@@ -1,188 +1,91 @@
...
-            defaultTags()
-            "$Tags.HTTP_STATUS.key" 500
-            "$Tags.HTTP_URL.key" "http://localhost:$port/$endpoint"
-            "$Tags.HTTP_METHOD.key" "GET"
-            "$Tags.SPAN_KIND.key" Tags.SPAN_KIND_SERVER
-            "$Tags.COMPONENT.key" "akka-http-server"
-            errorTags RuntimeException, errorMessage
-          }
-        }
+// FIXME: This doesn't work because we don't support bindAndHandle.
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/akka-http-10.0/src/test/groovy/AkkaHttpServerInstrumentationTest.groovy]: What is "This" referring to?

File: dd-java-agent/instrumentation/akka-http-10.0/src/test/scala/AkkaHttpTestWebServer.scala [hunk 2]
```diff
@@ -1,88 +1,43 @@
...
-import datadog.trace.agent.test.utils.PortUtils
-import datadog.trace.api.Trace
+import datadog.trace.agent.test.base.HttpServerTest.ServerEndpoint._
 
-import scala.concurrent.{Await, Future}
+import scala.concurrent.Await
 
-object AkkaHttpTestAsyncWebServer {
-  val port = PortUtils.randomOpenPort()
+// FIXME: This doesn't work because we don't support bindAndHandle.
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/akka-http-10.0/src/test/scala/AkkaHttpTestWebServer.scala]: Again, what exactly doesn't work?

File: dd-java-agent/instrumentation/akka-http-10.0/src/test/groovy/AkkaHttpServerInstrumentationTest.groovy (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/akka-http-10.0/src/test/groovy/AkkaHttpServerInstrumentationTest.groovy]: This is a known issue (and one customers have complained about) with our existing instrumentation: https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/instrumentation/akka-http-10.0/src/main/java/datadog/trace/instrumentation/akkahttp/AkkaHttpServerInstrumentation.java#L64-L69

I was mainly documenting the issue further in the tests making it easier if we have a chance to revisit this and try to get it working.  (How are you with akka streams?  Seems like we'd need to wrap the `Flow` to instrument at that lower level properly.)

File: dd-java-agent/instrumentation/akka-http-10.0/src/test/scala/AkkaHttpTestWebServer.scala (hunk 2 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/akka-http-10.0/src/test/scala/AkkaHttpTestWebServer.scala]: Same answer.. see above.

**Pull request merged**